pkg/git/provider_gitlab.go

```diff
@@ -47,22 +51,64 @@ func NewGitlabProvider(baseURL string) (Provider, error) {
 	}, nil
 }
 
-func (g *gitlab) Type() ProviderType {
-	return g.providerType
-}
-
 func (g *gitlab) BaseURL() string {
 	urlClone := *g.apiURL
 	urlClone.Path = ""
 	urlClone.RawQuery = ""
 	return urlClone.String()
 }
 
-func (g *gitlab) VerifyToken(ctx context.Context, tokenType TokenType, token string) error {
-	log.G(ctx).Debug("Skip verifying token for gitlab, to be implemented later")
+func (g *gitlab) SupportsMarketplace() bool {
+	return false
+}
+
+func (g *gitlab) Type() ProviderType {
+	return g.providerType
+}
+
+func (g *gitlab) VerifyRuntimeToken(ctx context.Context, token string) error {
+	return g.checkApiScope(ctx, token)
+}
+
+func (g *gitlab) VerifyUserToken(ctx context.Context, token string) error {
+	return g.checkReadRepositoryScope(ctx, token)
+}
+
+func (g *gitlab) checkApiScope(ctx context.Context, token string) error {
+	res, err := g.request(ctx, token, http.MethodPost, "projects")
```

Comment:
POST to `projects` without a body.
if it returns 400 - the token has "api" scope
otherwise - the token does not have the scope